### PR TITLE
[PL-48697] SMP user profiles

### PR DESCRIPTION
**Pull request by brian-f-harness**
Add a new SMP User profiles topic

For reviewers, a preview is available:

[User profiles](https://6631652ea9c5cb31eddc61e4--harness-developer.netlify.app/docs/self-managed-enterprise-edition/user-profiles)

## Merge requirements

- [x] Successful preview build.
- [x] No merge conflicts.
- [x] Up to date with main.
- [x] SME/Product review.
- [ ] Editorial/peer review.
- [x] No open change requests from reviewers.
- [n/a] Release notes/new features docs: Feature/version released to at least one prod environment.

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: docs/self-managed-enterprise-edition/user-profiles.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+There are four user profiles for Harness Self-Managed Enterprise Edition.
+
+- Demo
+- Small
+- Medium
+- Large
+
+## Profile size and module execution details
+
+| **Size** | **# of users** | **Parallel executions (CD)** | **Parallel executions (CI)** |**Parallel executions (STO)** | **Chaos experiments** |
```
Comment (bharatgoelharness) [inline docs/self-managed-enterprise-edition/user-profiles.md:18]: lets remove STO and CHAOS part as we didn't do the perf testing for these

File: docs/self-managed-enterprise-edition/user-profiles.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+This profile is used for demonstration purposes to allow you to test Harness Self-Managed Enterprise Edition before onboarding. This profile enables you to run up to six simultaneous executions across three modules and one Chaos Engineering experiment.
+
+### Demo user requirements
+
+Core CPU and memory requirements depend on the modules you use for demo purposes.
+
+- **CI and CD:** Minimum 3 core CPU and 14 Gi memory
+- **STO:** Additional 0.4 core CPU and 750 Mi memory
+- **Chaos Engineering:** Additional 0.4 core CPU and 750 Mi memory
```
Comment (bharatgoelharness) [inline docs/self-managed-enterprise-edition/user-profiles.md:35]: remove STO and CHAOS

File: docs/self-managed-enterprise-edition/user-profiles.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+   ```yaml
+    autoscaling:
+      enabled: true
+      minReplicas: 1
+      maxReplicas: 4
+      targetCPU: 80
+      targetMemory: 80
+   ```
+
+## Acquire a license
```
Comment (pratmit) [inline docs/self-managed-enterprise-edition/user-profiles.md:195]: What does this mean and what license?

Comment (brian-f-harness) [reply docs/self-managed-enterprise-edition/user-profiles.md:195]: We have this in another topic, so I used it here. Can remove it if it doesn't make sense here.
https://developer.harness.io/docs/self-managed-enterprise-edition/smp-supported-platforms#license-requirements

**Comment by brian-f-harness** [comment]
> Did anyone else approved this for technical accuracy?

@pratmit 
@bharatgoelharness reviewed and commented to remove STO and CE details that were in the initial draft since they haven't fully tested. I removed the info and requested another review. Will hold this PR until his approval.

File: docs/self-managed-enterprise-edition/user-profiles.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+   helm install my-release harness/harness-prod -n <namespace> -f your-override -f override-medium.yaml
+   ```
+
+   ```
+   helm upgrade my-release harness/harness-prod -n <namespace> -f your-override -f override-medium.yaml
+   ```
+
+##### Large
+
+```
```
Comment (brian-f-harness) [inline docs/self-managed-enterprise-edition/user-profiles.md:74]:
```suggestion
   ```
```

**Review by bharatgoelharness — DISMISSED** [review]

File: docs/self-managed-enterprise-edition/user-profiles.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+   helm install my-release harness/harness-prod -n <namespace> -f your-override -f override-small.yaml
+   ```
+
+   ```
+   helm upgrade my-release harness/harness-prod -n <namespace> -f your-override -f override-small.yaml
+   ```
+
+#### Medium
+
+```
```
Comment (brian-f-harness) [inline docs/self-managed-enterprise-edition/user-profiles.md:64]:
```suggestion
   ```
```

File: docs/self-managed-enterprise-edition/user-profiles.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+   helm install my-release harness/harness-prod -n <namespace> -f your-override.yaml -f override-demo.yaml
+   ```
+
+   ```
+   helm upgrade my-release harness/harness-prod -n <namespace> -f your-override -f override-demo.yaml
+   ```
+
+##### Small
+
+```
```
Comment (brian-f-harness) [inline docs/self-managed-enterprise-edition/user-profiles.md:54]:
```suggestion
   ```
```

**Review by SushrutHarness — APPROVED** [review]
LGTM

**Review by bharatgoelharness — APPROVED** [review]

**Pull request merged**